standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -8400,8 +8408,10 @@ public void commit_txn(CommitTxnRequest rqst) throws TException {
     }
     getTxnHandler().commitTxn(rqst);
     if (listeners != null && !listeners.isEmpty()) {
-      MetaStoreListenerNotifier.notifyEvent(listeners, EventType.COMMIT_TXN,
-          new CommitTxnEvent(rqst.getTxnid(), this));
+      if (!isHiveReplTxn) {
+        MetaStoreListenerNotifier.notifyEvent(listeners, EventType.COMMIT_TXN,
+                new CommitTxnEvent(rqst.getTxnid(), this));
+      }
```

Comment:
If you are skipping this event, should you go with next line:
Optional<CompactionInfo> compactionInfo = getTxnHandler().getCompactionByTxnId(rqst.getTxnid());

Reply:
if (!isHiveReplTxn) can be moved out